3-extra/1-guess-the-number-project/script.js

```diff
@@ -1,35 +1,122 @@
+  }
 
-  //If the users guess is lower than the random number print Number is too low, try again  (hint use .final-out class to print)
+  else
+  // If the users guess is lower than the random number print Number is too low, try again  
+  {
+                  message.innerText = "Number is too low, try again"
+                  --userTriesTaken; // subtract one from number of user's attempts
+  }
+  
+  if (userTriesTaken !== 0) {
+           //alert(triesOutputSelector)
+           triesOutputSelector.innerText = "Number of Tries: " + userTriesTaken;
+           return
+  }
 
-  //If the user has guessed the random number correctly print out the randomNumber with a message "Guess is correct. You win!"
+  // USER HAS LOST!
+  triesOutputSelector.innerText = "You lose, the number was " + randomNumber;
+  message.innerText = "";  
+  buttonGuessSelector.removeEventListener("click", guessNumber); // Disable Guess Button
+  guessButtonEnabled = false;
+  return;
 }
 
+
 // For this task we will be making a "New Game" button function which will reset our game,
 // Once the user clicks on this button the user will have new random number to guess
 // 1. Reset the values inside the body of the function
 // 2. Attach our new game button using an event listener to the .btnNewGame button
 function newGame() {
-  //Your code here
-  //Reset randomNumber
-  //Reset users input field
-  //Reset tries, and triesTaken by the user
+  // Reset randomNumber
+   randomNumber = Math.floor(Math.random() * 100 + 1); //  between 1-100 inclusive
+  // Reset users input field
+   inputsValuesSelector.value = "" //  reset the input field to blank 
+
+  // Reset tries, and triesTaken by the user
+   triesOutputSelector.innerText = "Number of Tries: " + maxNumberOfTries;
+   finalOutputSelector.innerText = "Guess a number between 1 and 100";
+   userTriesTaken = maxNumberOfTries;
+   if (!guessButtonEnabled)
+             buttonGuessSelector.addEventListener("click", guessNumber);
 }
 
-//keyboard exception
+/*
+According to MDN, the .keyCode property is deprecated:
+
+https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+// keyboard exception
 function keyBoardEvents(e) {
   if (e.keyCode === 13) {
     guessNumber();
   }
 }
+*/
+
+function keyBoardEvents(event) {
+  if (event.keyCode === 13) {
+        guessNumber();
+  }
+}
+
+
+function setup() {   
+   document.querySelector(".btnNewGame").addEventListener("click", newGame);
+   document.addEventListener("keypress", keyBoardEvents);
+   triesOutputSelector.innerText = "Number of Tries: " + maxNumberOfTries;
+   finalOutputSelector.innerText = "Guess a number between 1 and 100";
+   buttonGuessSelector.addEventListener("click", guessNumber);
+   userTriesTaken = maxNumberOfTries;
+   guessButtonEnabled = true;
+}
+
 
-document.querySelector(".btnGuess").addEventListener("click", guessNumber);
-document.addEventListener("keypress", keyBoardEvents);
+setup()
```

Comment:
I would put this in line 12 (after all the variable declarations). The reason is that, we read the codes from top to bottom. and by putting ``setup()`` in the beginning, it's easy to spot which function is called when this JS is executed